src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1997 @@
+                    set => throw null;
+                }
+            }
+            """;
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        comp.VerifyDiagnostics(
+            // (6,10): error CS9502: Cannot use 'OverloadResolutionPriorityAttribute' on this member.
+            //         [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, "OverloadResolutionPriority(1)").WithLocation(6, 10),
+            // (8,10): error CS9502: Cannot use 'OverloadResolutionPriorityAttribute' on this member.
+            //         [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, "OverloadResolutionPriority(1)").WithLocation(8, 10)
+        );
+
+        var c = comp.GetTypeByMetadataName("C")!;
+        var indexer = c.GetMember<PropertySymbol>("this[]");
+
+        Assert.Equal(0, indexer.OverloadResolutionPriority);
+        Assert.Equal(0, indexer.GetMethod.OverloadResolutionPriority);
+        Assert.Equal(0, indexer.SetMethod.OverloadResolutionPriority);
+    }
+
+    [Fact]
+    public void AppliedToIndexerGetterSetter_Metadata()
+    {
+        // Equivalent to:
+        // public class C
+        // {
+        //     public int this[object x]
+        //     {
+        //         [System.Runtime.CompilerServices.OverloadResolutionPriority(1)]
+        //         get => throw null;
+        //         [System.Runtime.CompilerServices.OverloadResolutionPriority(1)]
+        //         set => throw null;
+        //     }
+        //     public int this[string x]
+        //     {
+        //         get { System.Console.Write(1); return 1; }
+        //         set => System.Console.Write(2);
+        //     }
+        // }
+        var il = """
+            .class public auto ansi beforefieldinit C
+                extends [mscorlib]System.Object
+            {
+                .custom instance void [mscorlib]System.Runtime.CompilerServices.NullableContextAttribute::.ctor(uint8) = (
+                    01 00 01 00 00
+                )
+                .custom instance void [mscorlib]System.Runtime.CompilerServices.NullableAttribute::.ctor(uint8) = (
+                    01 00 00 00 00
+                )
+                .custom instance void [mscorlib]System.Reflection.DefaultMemberAttribute::.ctor(string) = (
+                    01 00 04 49 74 65 6d 00 00
+                )
+                // Methods
+                .method public hidebysig specialname 
+                    instance int32 get_Item (
+                        object x
+                    ) cil managed 
+                {
+                    .custom instance void System.Runtime.CompilerServices.OverloadResolutionPriorityAttribute::.ctor(int32) = (
+                        01 00 01 00 00 00 00 00
+                    )
+                    // Method begins at RVA 0x2050
```

Comment:
very nit: consider trimming other IL in tests to make them more compact. I typically just remove the irrelevant attributes and the the method preambles. Converting the multi-line signatures into single line is nice too.